### ISSUE TITLE
[BUG] Bundled HelmCharts in Errored State

### ISSUE THREAD

**Issue by deiberts86** [harvester/harvester#7087]
**Describe the bug**
(EDIT - Correction) Kubernetes CRD for "helmcharts" shows all status messages failed = false within the UI after a fresh install of Harvester. HelmChart bundles seems to point to jobs not succeeding. However, all jobs succeeded. 

**To Reproduce**
1. Install Harvester 1.4.0 (FRESH)
2. Execute command:
```sh
kubectl get helmcharts -A -o jsonpath='{range .items[*]}{"Namespace: "}{.metadata.namespace}{"\nName: "}{.metadata.name}{"\nStatus:\n"}{range .status.conditions[*]}{"  - Type: "}{.type}{"\n    Status: "}{.status}{"\n    Reason: "}{.reason}{"\n    Message: "}{.message}{"\n"}{end}{"JobName: "}{.status.jobName}{"\n\n"}{end}'
``` 
3. (EDIT - Correction) You will see that the jobs for each `helmChart` went through fine.
```txt
Namespace: cattle-logging-system
Name: rancher-logging
Status:
  - Type: JobCreated
    Status: True
    Reason: Job created
    Message: Applying HelmChart using Job cattle-logging-system/helm-install-rancher-logging
  - Type: Failed
    Status: False
    Reason: 
    Message: 
JobName: helm-install-rancher-logging

Namespace: cattle-monitoring-system
Name: rancher-monitoring
Status:
  - Type: JobCreated
    Status: True
    Reason: Job created
    Message: Applying HelmChart using Job cattle-monitoring-system/helm-install-rancher-monitoring
  - Type: Failed
    Status: False
    Reason: 
    Message: 
JobName: helm-install-rancher-monitoring

Namespace: harvester-system
Name: harvester-seeder
Status:
  - Type: JobCreated
    Status: True
    Reason: Job created
    Message: Applying HelmChart using Job harvester-system/helm-install-harvester-seeder
  - Type: Failed
    Status: False
    Reason: 
    Message: 
JobName: helm-install-harvester-seeder

Namespace: harvester-system
Name: pcidevices-controller
Status:
  - Type: JobCreated
    Status: True
    Reason: Job created
    Message: Applying HelmChart using Job harvester-system/helm-install-pcidevices-controller
  - Type: Failed
    Status: False
    Reason: 
    Message: 
JobName: helm-install-pcidevices-controller

Namespace: kube-system
Name: rke2-canal
Status:
  - Type: JobCreated
    Status: True
    Reason: Job created
    Message: Applying HelmChart using Job kube-system/helm-install-rke2-canal
  - Type: Failed
    Status: False
    Reason: 
    Message: 
JobName: helm-install-rke2-canal

Namespace: kube-system
Name: rke2-coredns
Status:
  - Type: JobCreated
    Status: True
    Reason: Job created
    Message: Applying HelmChart using Job kube-system/helm-install-rke2-coredns
  - Type: Failed
    Status: False
    Reason: 
    Message: 
JobName: helm-install-rke2-coredns

Namespace: kube-system
Name: rke2-ingress-nginx
Status:
  - Type: JobCreated
    Status: True
    Reason: Job created
    Message: Applying HelmChart using Job kube-system/helm-install-rke2-ingress-nginx
  - Type: Failed
    Status: False
    Reason: 
    Message: 
JobName: helm-install-rke2-ingress-nginx

Namespace: kube-system
Name: rke2-metrics-server
Status:
  - Type: JobCreated
    Status: True
    Reason: Job created
    Message: Applying HelmChart using Job kube-system/helm-install-rke2-metrics-server
  - Type: Failed
    Status: False
    Reason: 
    Message: 
JobName: helm-install-rke2-metrics-server

Namespace: kube-system
Name: rke2-multus
Status:
  - Type: JobCreated
    Status: True
    Reason: Job created
    Message: Applying HelmChart using Job kube-system/helm-install-rke2-multus
  - Type: Failed
    Status: False
    Reason: 
    Message: 
JobName: helm-install-rke2-multus

Namespace: rancher-vcluster
Name: rancher-vcluster
Status:
  - Type: JobCreated
    Status: True
    Reason: Job created
    Message: Applying HelmChart using Job rancher-vcluster/helm-install-rancher-vcluster
  - Type: Failed
    Status: False
    Reason: 
    Message: 
JobName: helm-install-rancher-vcluster
```
4.  Grab the jobs with kubectl and the `completions` column should be 1/1 for each job. Screenshot attached. 
```sh
kubectl get jobs -A 
```

5. Click on your User Icon within Harvester UI > Preferences > Click checkbox "Enable extension developer features" > Go back to Home Harvester Screen > Click on Support on bottom left corner > Click on "Access Embedded Rancher UI" > Click on "More Resources" on left hand pane > Click on "Helm" > Click on "Helm Charts
- You will see the "Errored" stated Helm Charts here as well.

**Expected behavior**
Helm charts shouldn't be in an error state if Jobs completed successfully within the UI itself. 

**Support bundle**
Will submit if requested. However, this has been observed numerous times with re-installs.

**Environment**
 - Harvester ISO version:  1.4.0
 - Bare Metal Dell PowerEdge R640

**Additional context**
Screenshots attached.  Sorry if the font is small in screenshots. 

![helmchart-errors](https://github.com/user-attachments/assets/4ad96b49-2637-4207-a75d-95eee54e79e6)
![Jobs Succeeded](https://github.com/user-attachments/assets/3acf86d5-f508-482c-aec6-83984b7af0e9)

**Comment by ihcsim** [comment]
The `HelmChart`s are correct. If there are no errors, the `Failed` condition has its [`Status` set to `False`](https://github.com/k3s-io/helm-controller/blob/5d91b5fef8482219a93ed6ccfc4ea026431966f8/pkg/controllers/chart/chart.go#L249-L260). OTOH, the UI might be wrong.

**Comment by deiberts86** [comment]
Yes, you're absolutely right.  Sorry, long day. The problem is the UI. The jobs completed and status seems to indicate that everything went through just fine.  I'll adjust my verbiage on the initial issue.

**Comment by deiberts86** [comment]
[supportbundle_63682b4f-c84a-4f8c-9b23-8f489add4777_2024-12-05T17-12-05Z.zip](https://github.com/user-attachments/files/18027207/supportbundle_63682b4f-c84a-4f8c-9b23-8f489add4777_2024-12-05T17-12-05Z.zip)

**Comment by CamrynCarter** [comment]
SURE-9478

**Comment by a110605** [comment]
I can reproduce this error by enabling some harvester addons.

The reason showing `Error` state is rancher UI framework consumes each row state by [this.metadata.state.name](https://github.com/rancher/dashboard/blob/df3dffca5622770e13b5b6ca3efb5e0e92335ad2/shell/plugins/steve/hybrid-class.js#L100-L102) which is `error`.

<img width="1496" alt="Screenshot 2024-12-24 at 4 04 39 PM" src="https://github.com/user-attachments/assets/fc0a2819-f7af-4bd7-a301-4336514361ee" />

`/v1/helm.cattle.io.helmcharts` should return correct `this.metadata.state` object. cc @ihcsim , @bk201 

Example of `Active` state of helm chart.

<img width="1496" alt="Screenshot 2024-12-24 at 4 43 25 PM" src="https://github.com/user-attachments/assets/4bdd0bab-8e0e-4608-b525-57b5b2da702c" />

**Comment by bk201** [comment]
@FrankYang0529 Please help check the helmchart resources.

**Comment by FrankYang0529** [comment]
This is upstream error. Nothing we can do in harvester side.

In helm controller, it introduced `Failed` condition.

https://github.com/k3s-io/helm-controller/blob/b40937ee695b33e80df2bbf4fa3437c8b9bce08d/pkg/apis/helm.cattle.io/v1/types.go#L73

In wrangler, it checks whether an object is error by checking `Failed` condition with `False` status.

https://github.com/rancher/wrangler/blob/abb0081704f3a7af13bf537a281937dadc0e6baa/pkg/summary/summarizers.go#L221-L235

https://github.com/rancher/wrangler/blob/abb0081704f3a7af13bf537a281937dadc0e6baa/pkg/summary/summarizers.go#L81-L83

That is why server returns `error: true` in API. To fix this, one of helm condition or wrangler error check needs to change.

**Comment by bk201** [comment]
Upstream issue: https://github.com/rancher/rancher/issues/48559

**Comment by innobead** [comment]
Added upstream labels, since this is a upstream issue instead.  cc @harvester/tpm

**Comment by Vicente-Cheng** [comment]
Already fixed on Rancher v2.10.2, REF: https://github.com/rancher/rancher/issues/48559#issuecomment-2882056033

close as fixed.
